library/ssl_misc.h

```diff
@@ -1434,4 +1437,12 @@ int mbedtls_ssl_tls13_write_sig_alg_ext( mbedtls_ssl_context *ssl,
 
 #endif /* MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL */
 
+static inline const void *mbedtls_get_supported_groups(const mbedtls_ssl_context *ssl){
+    #if defined(MBEDTLS_DEPRECATED_REMOVED) || !defined(MBEDTLS_ECP_C)
+        return ssl->conf->supported_groups;
+    #else
+        return ssl->handshake->group_list;
```

Comment:
We might want to check whether `ssl->handshake != NULL` here and return `NULL` if not, as a safeguard.

Reply:
Amended